diff --git a/types/jsx.d.ts b/types/jsx.d.ts
index 6628a0dfd87332a63faabeb92d1b304716271e15..af7c76f01e43e239252fce9d73628307bbc33444 100644
--- a/types/jsx.d.ts
+++ b/types/jsx.d.ts
@@ -240,13 +240,13 @@ export namespace JSX {
   }
 
   type RefCallback<T> = (el: T) => void;
-  type Ref<T> = T | RefCallback<T> | Ref<T>[];
+  type Ref<T> = T | RefCallback<T> | undefined | Ref<T>[];
 
   interface IntrinsicAttributes {
-    ref?: Ref<unknown> | undefined;
+    ref?: Ref<unknown>;
   }
   interface CustomAttributes<T> {
-    ref?: Ref<T> | undefined;
+    ref?: Ref<T>;
     children?: Element | undefined;
     $ServerOnly?: boolean | undefined;
   }
