diff --git a/dist/index.cjs b/dist/index.cjs
index 4f44e18794fb0281f22ba8980db96f79657d7fb2..d8d2c8d780798597ab5978ec234d7c80da0f6d5a 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -40,7 +40,7 @@ __export(src_exports, {
 module.exports = __toCommonJS(src_exports);
 var import_dom = require("@testing-library/dom");
 var import_solid_js = require("solid-js");
-var import_web = require("solid-js/web");
+var import_web = require("@solidjs/web");
 __reExport(src_exports, require("@testing-library/dom"), module.exports);
 if (!process.env.STL_SKIP_AUTO_CLEANUP) {
   if (typeof afterEach === "function") {
@@ -83,6 +83,7 @@ function render(ui, options = {}) {
     }
   }) : wrappedUi;
   const dispose = hydrate ? (0, import_web.hydrate)(routedUi, container) : (0, import_web.render)(routedUi, container);
+  (0, import_solid_js.flush)();
   mountedContainers.add({ container, dispose });
   const queryHelpers = (0, import_dom.getQueriesForElement)(container, queries);
   return {
@@ -111,6 +112,7 @@ function renderHook(hook, options) {
     }
     return [dispose2, (0, import_solid_js.getOwner)(), hook(...initialProps)];
   });
+  (0, import_solid_js.flush)();
   mountedContainers.add({ dispose });
   return { result, cleanup: dispose, owner };
 }
@@ -119,7 +121,7 @@ function renderDirective(directive, options) {
   return Object.assign(
     render(() => {
       const targetElement = options?.targetElement && (options.targetElement instanceof HTMLElement ? options.targetElement : typeof options.targetElement === "string" ? document.createElement(options.targetElement) : typeof options.targetElement === "function" ? options.targetElement() : void 0) || document.createElement("div");
-      (0, import_solid_js.onMount)(() => directive(targetElement, arg));
+      (0, import_solid_js.onSettled)(() => directive(targetElement, arg));
       return targetElement;
     }, options),
     { arg, setArg }
@@ -132,11 +134,15 @@ function testEffect(fn, owner) {
     context.fail = reject;
   });
   context.dispose = (0, import_solid_js.createRoot)((dispose) => {
-    (0, import_solid_js.onError)((err) => context.fail?.(err));
-    (owner ? (done) => (0, import_solid_js.runWithOwner)(owner, () => fn(done)) : fn)((result) => {
-      context.done?.(result);
-      dispose();
-    });
+    (0, import_solid_js.createErrorBoundary)(
+      () => {
+        (owner ? (done) => (0, import_solid_js.runWithOwner)(owner, () => fn(done)) : fn)((result) => {
+          context.done?.(result);
+          dispose();
+        });
+      },
+      (err) => { context.fail?.(err()); }
+    );
     return dispose;
   });
   return context.promise;
diff --git a/dist/index.js b/dist/index.js
index a2aee73db73861cc26129d42b9bdf20f89372e40..b647383cff52ef76d284bdf5641d0b1892cf0257 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2,15 +2,16 @@
 import { getQueriesForElement, prettyDOM } from "@testing-library/dom";
 import {
   createComponent,
+  createErrorBoundary,
   createRoot,
   createSignal,
+  flush,
   getOwner,
   lazy,
-  onError,
-  onMount,
+  onSettled,
   runWithOwner
 } from "solid-js";
-import { hydrate as solidHydrate, render as solidRender } from "solid-js/web";
+import { hydrate as solidHydrate, render as solidRender } from "@solidjs/web";
 export * from "@testing-library/dom";
 if (!process.env.STL_SKIP_AUTO_CLEANUP) {
   if (typeof afterEach === "function") {
@@ -53,6 +54,7 @@ function render(ui, options = {}) {
     }
   }) : wrappedUi;
   const dispose = hydrate ? solidHydrate(routedUi, container) : solidRender(routedUi, container);
+  flush();
   mountedContainers.add({ container, dispose });
   const queryHelpers = getQueriesForElement(container, queries);
   return {
@@ -81,6 +83,7 @@ function renderHook(hook, options) {
     }
     return [dispose2, getOwner(), hook(...initialProps)];
   });
+  flush();
   mountedContainers.add({ dispose });
   return { result, cleanup: dispose, owner };
 }
@@ -89,7 +92,7 @@ function renderDirective(directive, options) {
   return Object.assign(
     render(() => {
       const targetElement = options?.targetElement && (options.targetElement instanceof HTMLElement ? options.targetElement : typeof options.targetElement === "string" ? document.createElement(options.targetElement) : typeof options.targetElement === "function" ? options.targetElement() : void 0) || document.createElement("div");
-      onMount(() => directive(targetElement, arg));
+      onSettled(() => directive(targetElement, arg));
       return targetElement;
     }, options),
     { arg, setArg }
@@ -102,11 +105,15 @@ function testEffect(fn, owner) {
     context.fail = reject;
   });
   context.dispose = createRoot((dispose) => {
-    onError((err) => context.fail?.(err));
-    (owner ? (done) => runWithOwner(owner, () => fn(done)) : fn)((result) => {
-      context.done?.(result);
-      dispose();
-    });
+    createErrorBoundary(
+      () => {
+        (owner ? (done) => runWithOwner(owner, () => fn(done)) : fn)((result) => {
+          context.done?.(result);
+          dispose();
+        });
+      },
+      (err) => { context.fail?.(err()); }
+    );
     return dispose;
   });
   return context.promise;
